blockchain-agent/src/main/java/com/sphereon/alfresco/blockchain/agent/tasks/scheduled/VerifyPendingRegistrations.java

```diff
@@ -0,0 +1,53 @@
+package com.sphereon.alfresco.blockchain.agent.tasks.scheduled;
+
+import com.sphereon.alfresco.blockchain.agent.tasks.AlfrescoRepository;
+import com.alfresco.apis.handler.ApiException;
+import com.alfresco.apis.model.ResultNode;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import static com.sphereon.alfresco.blockchain.agent.model.AlfrescoBlockchainRegistrationState.PENDING_VERIFICATION;
+import static com.sphereon.alfresco.blockchain.agent.model.AlfrescoBlockchainRegistrationState.REGISTERED;
+
+@Component
+public class VerifyPendingRegistrations {
+    private static final int EXECUTION_RATE = 300000;
+
+    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(VerifyPendingRegistrations.class);
+
+    private final AlfrescoRepository alfrescoRepository;
+    private final VerifyPendingRegistrationsTask pendingRegistrationsTask;
+
+    public VerifyPendingRegistrations(final AlfrescoRepository alfrescoRepository,
+                                      final VerifyPendingRegistrationsTask pendingRegistrationsTask) {
+        this.alfrescoRepository = alfrescoRepository;
+        this.pendingRegistrationsTask = pendingRegistrationsTask;
+    }
+
+    @Scheduled(fixedRate = EXECUTION_RATE)
+    public synchronized void execute() {
+        try {
+            logger.info("Searching for documents with registration state " + PENDING_VERIFICATION);
+            this.alfrescoRepository.selectEntries(PENDING_VERIFICATION).forEach(rowEntry -> {
+                ResultNode entry = rowEntry.getEntry();
+                logger.info("Found document " + entry.getName() + " / " + entry.getId());
+                var contentHash = this.alfrescoRepository.hashEntry(entry.getId());
+                final var verifyResponse = this.pendingRegistrationsTask.verifyHash(contentHash);
+                if (verifyResponse.getRegistrationState() == REGISTERED) {
+                    logger.info("Updating state to pending for document " + entry.getName() + " / " + entry.getId());
```

Comment:
State is REGISTERED, but updating to Pending?

Reply:
Okay, now it starts to look like these log statements are just copy-pasted all the time =\

Reply:
Changed it to use string formatting, since it's anyway weird to first log what the state is going to be, and at the next line, retrieve the new state from somewhere else.